apps/internal/oauth/ops/accesstokens/accesstokens.go

```diff
@@ -106,14 +109,18 @@ func (c *Credential) JWT(authParams authority.AuthParams) (string, error) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	if c.Expires.Before(time.Now()) && c.Assertion != "" {
+	if c.Expires.After(time.Now()) {
```

Comment:
Can we stop caching assertions completely?  We should rely on token caching not on assertion caching. 

An assertion is not tied just to a certificate, to re-create it you need the `client_id` and the `tenant_id` (for normal AAD assertions). There will be other assertions like ManagedIdentity assertions, which are different.

Reply:
(we've had quite a few bugs because we were caching assertions)

Reply:
> Can we stop caching assertions completely? We should rely on token caching not on assertion caching.
> 

Sure, can it wait for a future PR? I'm making a tactical fix here to get certificate authentication working without breaking anything else. Removing or unexporting the `Expires` or `Assertion` fields entails rewriting unrelated tests and maybe some broad refactoring. I'm uncertain of the side effects but don't think they would be trivial.